tests/suites/test_suite_pk.function

```diff
@@ -69,6 +69,322 @@ size_t mbedtls_rsa_key_len_func( void *ctx )
  * END_DEPENDENCIES
  */
 
+/* BEGIN_CASE */
+void valid_parameters( )
+{
+    mbedtls_pk_context pk;
+    size_t len;
+
+    mbedtls_pk_init( &pk );
+
+    TEST_VALID_PARAM( mbedtls_pk_setup( &pk, NULL ) ==
+                      MBEDTLS_ERR_PK_BAD_INPUT_DATA );
+
+    TEST_VALID_PARAM( mbedtls_pk_get_bitlen( NULL ) == 0 );
+
+    TEST_VALID_PARAM( mbedtls_pk_get_len( NULL ) == 0 );
+
+    TEST_VALID_PARAM( mbedtls_pk_can_do( NULL, MBEDTLS_PK_NONE ) == 0 );
+
+    TEST_VALID_PARAM( mbedtls_pk_encrypt( &pk,
+                                          NULL, 0,
+                                          NULL, &len, 0,
+                                          rnd_std_rand, NULL ) ==
+                      MBEDTLS_ERR_PK_BAD_INPUT_DATA );
+
+    TEST_VALID_PARAM( mbedtls_pk_decrypt( &pk,
+                                          NULL, 0,
+                                          NULL, &len, 0,
+                                          rnd_std_rand, NULL ) ==
+                      MBEDTLS_ERR_PK_BAD_INPUT_DATA );
+
+#if defined(MBEDTLS_PK_PARSE_C)
+    TEST_VALID_PARAM( mbedtls_pk_parse_key( &pk, NULL, 0, NULL, 1 ) ==
+                      MBEDTLS_ERR_PK_KEY_INVALID_FORMAT );
+
+    TEST_VALID_PARAM( mbedtls_pk_parse_public_key( &pk, NULL, 0 ) ==
+                      MBEDTLS_ERR_PK_KEY_INVALID_FORMAT );
+#endif /* MBEDTLS_PK_PARSE_C */
+
+#if defined(MBEDTLS_PK_WRITE_C)
+    TEST_VALID_PARAM( mbedtls_pk_write_key_der( &pk, NULL, 0 ) ==
+                      MBEDTLS_ERR_PK_FEATURE_UNAVAILABLE );
+
+    TEST_VALID_PARAM( mbedtls_pk_write_pubkey_der( &pk, NULL, 0 ) ==
+                      MBEDTLS_ERR_PK_FEATURE_UNAVAILABLE );
+#endif /* MBEDTLS_PK_WRITE_C */
+}
+/* END_CASE */
+
+/* BEGIN_CASE depends_on:MBEDTLS_CHECK_PARAMS:!MBEDTLS_PARAM_FAILED_ALT */
+void invalid_parameters( )
+{
+    size_t len;
+    unsigned char *null_buf = NULL;
+    unsigned char buf[1];
+    unsigned char *p = buf;
+    char str[1] = {0};
+    mbedtls_pk_context pk;
+    void *options = buf;
+
+    mbedtls_pk_init( &pk );
+
+    TEST_INVALID_PARAM( mbedtls_pk_init( NULL ) );
+
+#if defined(MBEDTLS_ECDSA_C) && defined(MBEDTLS_ECP_RESTARTABLE)
+    TEST_INVALID_PARAM( mbedtls_pk_restart_init( NULL ) );
+
+    TEST_INVALID_PARAM( mbedtls_pk_restart_free( NULL ) );
```

Comment:
This should be a positive test in `valid_parameters()`.

Reply:
fixed